s/api/services/v1/comment-writing-service.ts

```diff
@@ -102,41 +102,43 @@ export const makeCommentWritingService = asServiceProvider(({
 		})
 	},
 
-	async archiveComment(rawId: string): Promise<void> {
-		const id = enforceValidation(rawId, validateId)
+	async archiveComments(rawIds: string[]): Promise<void> {
+		const idArr = enforceValidation(rawIds, validateIdArray)
 
 		if (!user)
 			throw new renraku.ApiError(403, "cannot archive, not logged in")
 
-		const specificComment = await database.tables.comments.readOne(
-			find({id: dbmage.Id.fromString(id)})
-		)
-
-		if (!specificComment)
-			throw new renraku.ApiError(404, "cannot archive, comment not found")
-
-		const userIsTheAuthor = user.id === specificComment.authorId.string
-		const userHasAdminRights = user.permissions.canArchiveAnyComment
-		const userIsAllowedToArchive = userIsTheAuthor || userHasAdminRights
-
-		if (!userIsAllowedToArchive) 
-			throw new renraku.ApiError(403, "forbidden, must be author or admin to archive a comment")
-
-		await database.transaction(async({tables}) => {
-			await tables.comments.update({
-				...find({id: dbmage.Id.fromString(id)}),
-				write: {archived: true},
-			})
-
-			// TODO dbmage currently requires this count.. but shouldn't
-			const scoreCount = await tables.scores.count({
-				...find({commentId: dbmage.Id.fromString(id)}),
-			})
-			if (scoreCount)
-				await tables.scores.update({
-					...find({commentId: dbmage.Id.fromString(id)}),
+		for (let index = 0; index < idArr.length; index++) {
+			const id = idArr[index];
+			const specificComment = await database.tables.comments.readOne(
+				find({id: dbmage.Id.fromString(id)})
+			)
+			if (!specificComment)
+				throw new renraku.ApiError(404, "cannot archive, comment not found")
+
+				const userHasAdminRights = user.permissions.canArchiveAnyComment
+				const userIsAllowedToArchive = userHasAdminRights
+		
+			if (!userIsAllowedToArchive) 
+				throw new renraku.ApiError(403, "forbidden, must be author or admin to archive a comment")
+
+			await database.transaction(async({tables}) => {
+				await tables.comments.update({
+					...find({id: dbmage.Id.fromString(id)}),
```

Comment:
we really need to run a single database call, using dbmage.findAll, to archive all the comments in one call

Reply:
the ```findAll``` method keeps throwing errors. for your convenience, i have attached screenshots and have commented the code for you to review.

![Screenshot from 2022-05-30 14-11-06](https://user-images.githubusercontent.com/62189260/171016236-e5481121-6b55-4d8d-bc4a-bcf1b2592b4c.png)
![Screenshot from 2022-05-30 14-11-26](https://user-images.githubusercontent.com/62189260/171016288-b53d794a-eb01-4eb3-853e-fb68c31e1037.png)
![Screenshot from 2022-05-30 14-13-33](https://user-images.githubusercontent.com/62189260/171016322-76dbbee5-c87e-4a2f-baf4-bea6ca4217f5.png)
![Screenshot from 2022-05-30 14-13-39](https://user-images.githubusercontent.com/62189260/171016343-8dea5427-ab72-422a-a448-b803d6f6594d.png)

Reply:
ah, the typescript error is misleading.

the problem is actually that we need to convert the id to binary when using it for a database query.

```ts
const ids = idArr.map(id => dbmage.Id.fromString(id))

await database.tables.comments.update({
    ...dbmage.findAll(ids, id => ({id})),
    write: {archived: true},
})
```